src/syntax-highlight/syntax-highlight.js

```diff
@@ -69,19 +69,38 @@ export default function syntaxHighlight(diff, afterWordDiff) {
     }
 }
 
-function syntaxHighlightSourceCodeLines($diff) {
-    const sourceLines = [
-        ...$diff.find('pre:not([class*=language]), pre:has(ins), pre:has(del)'),
-    ]
-
-    sourceLines.forEach(preElement => {
-        if (!preElement.firstChild.$$rbb_isSyntaxHighlighted) {
-            Prism.highlightElement(preElement)
-            preElement.classList.add('__rbb_syntax_highlight')
-            // eslint-disable-next-line camelcase
-            preElement.firstChild.$$rbb_isSyntaxHighlighted = true
-        }
-    })
+async function syntaxHighlightSourceCodeLines($diff, querySelector) {
+    const sourceLines = [...$diff.find(querySelector)]
+
+    const promises = sourceLines.map(
+        preElement =>
+            new Promise((resolve, reject) => {
+                const { classList, firstChild, innerText } = preElement
+                if (
+                    firstChild.$$rbb_isSyntaxHighlighted &&
```

Comment:
went bitbucket re-renders a line with word-diff it changes the content, it leaves the class and property there, so I check for childElementCount (tokens) to see if it's just plain text or it was highlighted